pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoWriterTools.java

```diff
@@ -462,23 +457,22 @@ public static <T> void writeOneRepeatedMessage(final WritableSequentialData out,
      * @param out The data output to write to
      * @param field the descriptor for the field we are writing
      * @param message the message to write
-     * @param writer method reference to writer for the given message type
-     * @param sizeOf method reference to sizeOf measure method for the given message type
+     * @param codec the codec for the given message type
      * @throws IOException If a I/O error occurs
      * @param <T> type of message
      */
     private static <T> void writeMessageNoChecks(final WritableSequentialData out, final FieldDefinition field,
-            final T message, final ProtoWriter<T> writer, final ToIntFunction<T> sizeOf) throws IOException {
+            final T message, final Codec<T> codec) throws IOException {
         // When not a oneOf don't write default value
         if (field.oneOf() && message == null) {
             writeTag(out, field, WIRE_TYPE_DELIMITED);
             out.writeVarInt(0, false);
         } else if (message != null) {
             writeTag(out, field, WIRE_TYPE_DELIMITED);
-            final int size = sizeOf.applyAsInt(message);
+            final int size = codec.measureRecord(message);
             out.writeVarInt(size, false);
```

Comment:
Not related to the current changes, but still a good question. If an object contains all default fields, its size is 0 (default values are not written). Should such an object be written at all, with 0 length, or omitted?

Reply:
Omitted I think, though if that is different to know then surprised it has not come up in comparative testing with ProtoC